### ISSUE TITLE
What is correct way to start ASP.NET core application, when SQL connection fails

### ISSUE THREAD

**Issue by hokpe** [dotnet/efcore#17576]
I have a ASP.NEt core 2.1 server that is started on-demand. So when someone enters to the web page. My SQL server is in the other location than the web server and there is cases when the SQL server is not up and runnig. So I would need to gracefully start the web server without the SQL server. But so that it of course tries to connect to the SQL server first.

This was working somehow in the ASP.NET Core 1.1, but now with 2.1 I'm getting the exception InvalidOperationException: No database provider has been configured for this DbContext.

I'm using dynamic connection string and I have a function that either returns the connection string, of some other info (like if failure is detected already before.) Code example and same question in the stackoverflow as well:
https://stackoverflow.com/questions/56656167/what-is-correct-way-to-start-asp-net-core-application-when-sql-connection-fails

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a47bf610-692d-9d40-d85f-aec1e2a87a07
* Version Independent ID: 3c94634a-4b17-bc13-7436-0c61aea03ab6
* Content: [Configuring a DbContext - EF Core](https://docs.microsoft.com/en-us/ef/core/miscellaneous/configuring-dbcontext#feedback)
* Content Source: [entity-framework/core/miscellaneous/configuring-dbcontext.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/miscellaneous/configuring-dbcontext.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @rowanmiller
* Microsoft Alias: **divega**

**Comment by ajcvickers** [comment]
@hokpe Can you please post a small, runnable project/solution or complete code listing that demonstrates the behavior you are seeing?

**Comment by hokpe** [comment]
Hi ajcvickers. Thank you for your interest. Today I tried to make such an example. And I added one by one all my used features to the example project generated by the visual studio. And finally I had all the possible parts in it. And it was still working. Then I returned to my actual project and tested the same thing there. And it is working as well. I have updated the .NET core SDK and .NET core runtimes for other reasons, but I didn't test is this working now. So only conclusion is that those updates has fixed this as well.
